examples/pytorch/ogc/ogc.py

```diff
@@ -0,0 +1,116 @@
+import argparse
+import time
+
+import scipy.sparse as sp
+
+import torch
+import torch.nn.functional as F
+
+from dgl import AddSelfLoop
+from dgl.data import CiteseerGraphDataset, CoraGraphDataset, PubmedGraphDataset
+from utils import (
+    LinearNeuralNetwork,
+    sparse_mx_to_torch_sparse_tensor,
+    symmetric_normalize_adjacency,
+)
+
+
+# Training settings.
+decline = 0.9  # the decline rate
+lr_sup = 0.001  # the learning rate for supervised loss
+lr_clf = 0.5  # the learning rate for the used linear classifier
+beta = 0.1  # the moving probability that a node moves to its neighbors
+max_sim_rate = 0.995  # the max label prediction similarity between iterations
+max_patience = 2  # the tolerance for consecutively similar test predictions
+
+
+def update_embeds(embeds, graph, label_idx_mat):
```

Comment:
This is a part of the OGC model, right?  If so, you could put it in an `OGC` NN module.

Reply:
Thanks. As suggested, we have incorporated this function into our OGC model in the 'ogc.py' file.